models/basgra/src/Makevars

```diff
@@ -0,0 +1,40 @@
+# PEcAn BASGRA -- Makevars
+# Author: Istem Fer, Alexey Shiklomanov
+
+PKG_LIBS = $(FLIBS) 
+
+
+SOURCES = \
+    model/parameters_site.f90 \
+    model/parameters_plant.f90 \
+    model/environment.f90 \
+    model/resources.f90 \
+    model/soil.f90 \
+    model/plant.f90 \
+    model/set_params.f90 \
+    model/BASGRA.f90 
+
+
+OBJECTS = \
+    parameters_site.o \
+    parameters_plant.o \
+    environment.o \
+    resources.o \
+    soil.o \
+    plant.o \
+    set_params.o \
+    BASGRA.o 
+    
+
+.PHONY: all clean
+
+all : $(SHLIB) 
+
+$(SHLIB) : $(OBJECTS) 
+
+$(OBJECTS) :  $(SOURCES)
+	$(F77) -x f95-cpp-input -fPIC -O3 -c -fdefault-real-8 $(SOURCES) 
```

Comment:
I would probably write this as:

```
PKG_FFLAGS = -x f95-cpp-input -fdefault-real-8
$(OBJECTS) : $(SOURCES)
```

...and let R's default make macros work out the rest. Following [these recommendations](https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Using-Makevars).

Reply:
this gives me `g++: error: *o: No such file or directory` kind of errors

Reply:
Oh, I think that's probably because the compiled `.o` files in `OBJECTS` are placed in the same directory as their source files. So you need to prefix all of your `OBJECTS` with `model/`.

Reply:
no, they're not being created at all in neither directories. For some reason it starts with g`++ -shared` and looks for `.o` files if I arrange commands like this. I think it should start with `gfortran -x f95-cpp-input -fPIC -O3 -c -fdefault-real-8 `, look for source files and create `.o`s first

Reply:
this was the combination that worked for me
```
all : $(SHLIB) 

$(SHLIB) : $(SOURCES) 

PKG_FFLAGS = -x f95-cpp-input -fPIC -O3 -c -fdefault-real-8 $(SOURCES) 

$(OBJECTS) : $(SOURCES)
	$(F77) $(PKG_FFLAGS)
```

Reply:
Huh, interesting. I thought R was smart enough to figure out how to do this without the fully explicit rule, but I guess not. If it works, let's leave it as is, and potentially revisit if it causes anyone problems.